science/apbs/Portfile

```
@@ -20,64 +17,86 @@ long_description        APBS is a software package for the numerical \
-                        -DCMAKE_C_COMPILER_ARG1:STRING="-I${worksrcpath}/include ${configure.cflags}" \
-                        -DCMAKE_CXX_COMPILER_ARG1:STRING="-I${worksrcpath}/include ${configure.cxxflags} -stdlib=libc++"
-
-# otherwise obsolete subport installation will fail on patch phase and not get to pre-configure error about replacement
-if {${subport} eq ${name}} {
-    patchfiles              patch-CMakeLists.txt.diff patch-Eigen-include-path.diff patch-contrib-tabipb-CMakeLists.txt.diff patch-pbsolvercz.cpp.diff
-    if {[mpi_variant_isset]} {
-        patchfiles-append   patch-tests-apbs_tester.py.diff
+version                 3.4.1
+homepage                https://www.poissonboltzmann.org
+revision                0
+worksrcdir              ${workpath}/apbs-${version}
+
+extract.mkdir           no
+master_sites            https://github.com/Electrostatics/FETK/archive/refs/tags/ \
+                        https://github.com/Electrostatics/apbs/archive/refs/tags/
+distfiles               v${version}.tar.gz 1.9.3.tar.gz 
+checksums               1.9.3.tar.gz \
+                        rmd160  25cf57754a0b817e76e19b8de93617e554302661 \
+                        sha256  2ce7ab04cba4403f4208c3ecf1c81a0a18aae6a77d22da0ffa5f64c2da7c6f28 \
+                        size    27757348 \
+                        v${version}.tar.gz \
+                        rmd160  17de250c3887ff68f89a873c5a0be19f6ab38813 \
+                        sha256  644e6246fd37c9dd4172fc7be1763337082ab8584dbcd53e738c831b62c89bb2 \
+                        size    85806314
+
+# Test support requires python3 but the
+# building the apbs python module with swig
+# is unnecessary for apbs's use in pymol.
+platform darwin {
+   if {${os.major} < 13} {
+       depends_test      port:python312
     }
 }
 
+depends_lib             port:maloc \
+                        port:superlu \
+                        port:SuiteSparse
+
+configure.optflags      -O3 -ffast-math -fcommon -Wno-error=implicit-int -Wno-error=incompatible-function-pointer-types 
+# Follow Debian's approach of not building those
+# features like BEM, GEOFLOW or PBAM that only
+# exist in apbs's github without tagged releases yet.
+configure.args-append   -DAPBS_STATIC_BUILD=ON \
+                        -DBUILD_SHARED_LIBS=OFF \
+                        -DBUILD_TOOLS=ON \
+                        -DENABLE_OPENMP:BOOL=OFF \
+                        -DENABLE_BEM=OFF \
+                        -DENABLE_GEOFLOW=OFF \
+                        -DENABLE_PBAM=OFF \
+                        -DBUILD_MALOC=OFF \
+                        -DBUILD_SUPERLU=OFF \
+                        -DBUILD_SHARED_LIBS=ON \
+                        -DFETCHCONTENT_SOURCE_DIR_FETK=${worksrcpath}/FETK-1.9.3 \
+                        -DCMAKE_C_COMPILER_ARG1:STRING="${configure.cflags} -I${worksrcpath}/include" \
+                        -DCMAKE_CXX_COMPILER_ARG1:STRING="${configure.cxxflags} -I${worksrcpath}/include -stdlib=libc++" \
```

You should not assume which stdlib is in use. Use the `${configure.cxx_stdlib}` variable to get the stdlib the user wants, except that you don't need to do that because MacPorts has already put the right `-stdlib` flag into `${configure.cxxflags}` for you.

`-I` flags are preprocessor flags and belong in `${configure.cppflags}` so, earlier in the Portfile, I would use:

```
configure.cppflags-append -I${worksrcpath}/include
```

Since cmake doesn't support CPPFLAGS, the cmake portgroup puts `${configure.cppflags}` into `${configure.cflags}` and `${configure.cxxflags}` for you, so then you only have to:

```suggestion
                        -DCMAKE_C_COMPILER_ARG1:STRING="${configure.cflags}" \
                        -DCMAKE_CXX_COMPILER_ARG1:STRING="${configure.cxxflags}" \
```

---

Now that I take another look at this, I wonder why `${configure.cflags}` and `${configure.cxxflags}` need to be passed in this way. Does this build system not accept the usual way that the cmake portgroup already passes those in?

---

Removed unnecessary CMAKE_C_COMPILER_ARG1 and CMAKE_CXX_COMPILER_ARG1 cmake defines.